testing/sdk_metrics.py

```diff
@@ -71,10 +72,13 @@ def get_metrics(package_name, service_name, task_name):
 
 
 def wait_for_any_metrics(package_name, service_name, task_name, timeout):
-    def metrics_exist():
+    @retrying.retry(
+        wait_fixed=5000,
+        stop_max_delay=timeout*1000)
+    def wait_for_metrics_to_exist():
         log.info("verifying metrics exist for {}".format(service_name))
         service_metrics = get_metrics(package_name, service_name, task_name)
         # there are 2 generic metrics that are always emitted
-        return len(service_metrics) > 2
+        assert len(service_metrics) > 2
```

Comment:
Will this not retry infinitely?

Reply:
See other comments above explaining this API